src/main/java/com/googlecode/zohhak/internal/Executor.java

```diff
@@ -1,17 +1,20 @@
 package com.googlecode.zohhak.internal;
 
+import com.googlecode.zohhak.api.Configuration;
+import com.googlecode.zohhak.api.backend.ParameterCalculator;
 import com.googlecode.zohhak.internal.coercing.CoercingService;
 import com.googlecode.zohhak.internal.model.ConfigurationBuilder;
-import com.googlecode.zohhak.internal.model.SingleTestMethod;
 import com.googlecode.zohhak.internal.parsing.ParsingService;
 
-public final class Executor {
+import java.lang.reflect.Method;
+
+public final class Executor implements ParameterCalculator {
 
 	ConfigurationResolver configurationResolver;
 	CoercingService coercingService;
 	ParsingService parsingService;
-	
-	public Object[] calculateParameters(SingleTestMethod singleTestMethod, String parametersLine) {
+
+	public Object[] calculateParameters(String parametersLine, Method testMethod, Configuration... additionalConfiguration) {
```

Comment:
The 3rd parameter is used, the additionalConfiguration is passed into configurationResolver.calculateConfiguration(testMethod, additionalConfiguration)